### PR TITLE
demo: Add QPL demo

### PR DESCRIPTION
**Pull request by ozhuraki**
Issues:
https://github.com/intel/qpl/pull/12

Init tests require privileged mode, low level examples do not.

```
[==========] 7 tests from 2 test suites ran. (71 ms total)
[  PASSED  ] 7 tests.

ll_canned_mode_example
Content was successfully compressed and decompressed.
Compressed size: 17
0
ll_compression_example
Content was successfully compressed and decompressed.
Compressed size: 20
0
ll_crc64_example
CRC64 was performed successfully.
0
ll_expand_example
Expand was performed successfully.
0
ll_extract_example
Extract was performed successfully.
0
ll_scan_example
Scan was performed successfully.
0
ll_scan_for_unique_value_example
Scan was performed successfully.
0
ll_scan_range_example
Scan range was performed successfully.
0
ll_select_example
Select was performed successfully.
0
ll_serialization_example
Huffman table was successfully serialized and deserialized.
Serialized size: 31251
0

real    0m0.343s
user    0m0.071s
sys     0m0.036s
```

### CODE REVIEW

File: demo/iaa-qpl-demo/Dockerfile [hunk 1]
```diff
@@ -12,39 +12,39 @@
...
 # limitations under the License.
 
-FROM ubuntu:20.04 AS builder
+FROM ubuntu:22.04 AS builder
 
-RUN apt update && DEBIAN_FRONTEND=noninteractive TZ="Etc/UTC" \
-    apt install -y --no-install-recommends \
-    g++ clang nasm cmake make git ca-certificates uuid-dev \
-    gcc autoconf automake libtool pkg-config libjson-c-dev curl zlib1g-dev
+RUN apt update && apt install -y --no-install-recommends \
```
Comment (eero-t) [inline demo/iaa-qpl-demo/Dockerfile:17]: Why you're removing `DEBIAN_FRONTEND=noninteractive`?

Also, please run the Dockerfile through `hadolint`.

PS. Feel free to use `.hadolint.yaml` config file like this:
```
ignored:
# use Docker "--no-cache" to get security updates instead
# of tracking and constantly updating pinned package versions:
- DL3008
```

File: demo/iaa-qpl-demo/Dockerfile [hunk 2]
```diff
@@ -27,25 +26,24 @@ RUN curl -fsSL "$ACCEL_CONFIG_DOWNLOAD_URL" -o accel-config.tar.gz && echo "$ACC
...
     make install
 
-RUN git clone --recursive --depth 1 --branch v1.0.0 \
-    https://github.com/intel/qpl.git && \
+RUN git clone --recursive --depth 1 --branch v1.0.0 https://github.com/intel/qpl.git && \
     mkdir qpl/build && cd qpl/build && \
     sed -i '10i #include <stdexcept>' ../tools/benchmarks/include/types.hpp && \
     cmake -DLOG_HW_INIT=ON .. && \
-    cmake --build . --target install
+    make install
```
Comment (mythi) [inline demo/iaa-qpl-demo/Dockerfile:36]: QPL build instructions suggested to use `cmake --build ...` so I moved to use it

**Comment by mythi** [comment]
@ozhuraki we can close this, right?

**Comment by mythi** [comment]
> @ozhuraki we can close this, right?

Closing since we have #1286

**Pull request closed without merging**